storage/oqgraph/mysql-test/oqgraph/general-Aria.result

```diff
@@ -195,6 +195,170 @@ from	to
 SELECT linkid as `from`, destid as `to` FROM graph where latch='0' and destid = 10;
 from	to
 12	10
+# Leaves search tests
+SELECT * FROM graph WHERE latch = 'leaves' AND origid = 1;
+latch	origid	destid	weight	seq	linkid
+SELECT * FROM graph WHERE latch = 'leaves' AND origid = 2;
+latch	origid	destid	weight	seq	linkid
+SELECT * FROM graph WHERE latch = 'leaves' AND origid = 3;
+latch	origid	destid	weight	seq	linkid
+SELECT * FROM graph WHERE latch = 'leaves' AND origid = 4;
+latch	origid	destid	weight	seq	linkid
+SELECT * FROM graph WHERE latch = 'leaves' AND origid = 5;
+latch	origid	destid	weight	seq	linkid
+leaves	5	NULL	1	1	7
+SELECT * FROM graph WHERE latch = 'leaves' AND origid = 6;
+latch	origid	destid	weight	seq	linkid
+leaves	6	NULL	2	1	7
+SELECT * FROM graph WHERE latch = 'leaves' AND origid = 7;
+latch	origid	destid	weight	seq	linkid
+leaves	7	NULL	0	1	7
+SELECT * FROM graph WHERE latch = 'leaves' AND origid = 8;
+latch	origid	destid	weight	seq	linkid
+SELECT * FROM graph WHERE latch = 'leaves' AND origid = 9;
```

Comment:
The test case looks fine to me except for node 9. Node 9 is an isolated node. Wouldn't node 9 qualify as both a "root" and a "leaf" node?

Reply:
Technically, no. "root" and "leaf" would be defined as nodes with only outgoing or only incoming edges. Node 9 has both, hence doesn't qualify as a leaf/root node.

Reply:
I assume you meant node 9 has neither (no incoming/no outgoing edges).

So the definition is that it must have > 0 outgoing or > 0 incoming edges. If both are 0, it is neither a leaf nor a root. Sounds ok with me, although is worthy of being specified somewhere. Not sure where we ought to document this. (somewhere in the KnowledgeBase, probably https://mariadb.com/kb/en/library/oqgraph-overview/)

Reply:
Node 9 is set up using
```
INSERT INTO graph_base(from_id, to_id) VALUES (9,9);
```
That gives it one edge, which is both outgoing (to node 9) and incoming (to node 9)

Every row in an OQGRAPH table maps an edge. It's not possible to have a node without at least one edge in that structure.

The documentation definitely needs some adjustments once this is in. I'll try to get to that ASAP :)